persist/sqlite/metrics_test.go

```diff
@@ -439,6 +440,81 @@ func TestV2Metrics(t *testing.T) {
 	assertMetrics(metricsGenesis)
 }
 
+func TestDifficultyMetrics(t *testing.T) {
+	n := newTestChain(t, false, nil)
+
+	tests := []struct {
+		name             string
+		start, end, step uint64
+		err              string
+	}{
+		{
+			name:  "Basic range",
+			start: 0, end: 2, step: 1,
+		},
+		{
+			name:  "Step of 2",
+			start: 0, end: 4, step: 2,
+		},
+		{
+			name:  "Empty range",
+			start: 5, end: 5, step: 1,
+		},
+		{
+			name:  "Start equals end",
+			start: 5, end: 5, step: 1,
+		},
+		{
+			name:  "Invalid range",
+			start: 5, end: 3, step: 1,
+			err: "start height 5 cannot be greater than end height 3",
+		},
```

Comment:
(nit) might be good to add a test case for when the step doesn't cleanly divide the length and another one for step = 0.